swap/src/main.rs

```diff
@@ -37,13 +36,36 @@ pub const PORT: u16 = 9876; // Arbitrarily chosen.
 pub const ADDR: &str = "127.0.0.1";
 pub const BITCOIND_JSON_RPC_URL: &str = "127.0.0.1:8332";
 
+#[cfg(feature = "tor")]
+use swap::tor::{AuthenticatedConnection, UnauthenticatedConnection};
+#[cfg(feature = "tor")]
+use torut::onion::TorSecretKeyV3;
+#[cfg(feature = "tor")]
+pub const TOR_PORT: u16 = PORT + 1;
+
 #[tokio::main]
 async fn main() -> Result<()> {
     let opt = Options::from_args();
 
     trace::init_tracing(LevelFilter::Debug)?;
 
-    let addr = format!("/ip4/{}/tcp/{}", ADDR, PORT);
+    let (addr, _tor_connection) = {
+        #[cfg(feature = "tor")]
+        {
+            let tor_secret_key = TorSecretKeyV3::generate();
```

Comment:
If we generate a new address for every instance (and we currently use one instance per swap) that means that Alice must share this address with Bob _after_ she has started the node? Is this what we want?

Reply:
There are multiple solutions to this. Simplest is: yes, Alice needs to start first and then share it with Bob. This is fine as we have a single-trade-and-terminate application. 

In a later version we can store this key, derive it from a seed, etc.